graph/src/data/subgraph/schema.rs

```diff
@@ -199,6 +200,22 @@ impl SubgraphDeploymentEntity {
         }
     }
 
+    /// Return true if the deployment with the given id is fully synced,
+    /// and return false otherwise. Errors from the store are passed back up
+    pub fn is_synced(
+        store: &Store,
+        id: &SubgraphDeploymentId,
+    ) -> Result<bool, QueryExecutionError> {
+        let filter = EntityFilter::new_equal("id", id.0.clone());
+        let entity = store.find_one(SubgraphDeploymentEntity::query().filter(filter))?;
+        entity
+            .map(|entity| match entity.get("synced") {
+                Some(Value::Bool(true)) => Ok(true),
+                _ => Ok(false),
+            })
+            .unwrap_or(Ok(false))
+    }
+
```

Comment:
We had an unwritten rule here of avoiding using references to the `Store` -- no side effects, pure functions only in this file

Reply:
Is there a better place to put something like that? I am a little hesitant to just stick that into `create_subgraph_version` since it is less clear when it's just inlined.

Reply:
There's some similar stuff as methods on the Store trait, see https://github.com/graphprotocol/graph-node/blob/master/graph/src/components/store.rs#L419
Could also just use a function in this file.

The Store trait is getting a bit heavy, I'm not sure what the best solution is there. We may need another component or something

Reply:
I moved this to the Store trait.